### ISSUE TITLE
Window resize behavior is unexpected 

### ISSUE THREAD

**Issue by SimonMaracine** [pygame/pygame#1624]
I'll provide the code and then explain my problem.

```
import pygame

pygame.init()
window = pygame.display.set_mode((800, 600), pygame.RESIZABLE)

running = True
while running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.ACTIVEEVENT:
            print(event)
        elif event.type == pygame.VIDEORESIZE:
            print(event)
        elif event.type == pygame.KEYDOWN:
            print(f"width={window.get_width()}, height={window.get_height()}")

    window.fill((255, 0, 0))
    pygame.display.flip()
```

When I resize the window, the area of the surface returned by display.set_mode(), i. e. the red area, remains the same, i. e. `window.fill((255, 0, 0))` is still 800 by 600 and doesn't fill the entire window. The remaining area is completely black, in fact, you cannot draw on that area. **Is this the correct behaviour?** If I press a key, it still prints the old 800 by 600 resolution, even if the window was resized.

I also discovered a weird thing:
On Linux (Ubuntu 18.04.4), the cursor generates an ACTIVEEVENT only when it crosses the old area of 800x600, not the window area. But on Windows, it generates the event when it crosses the window (which seems the right thing). I have tested all of this with pygame 1.9.6, Python 3. The resizing thing I tested also on pygame 1.9.4 and 1.9.5 and it's the same.

What I wish is to draw on the window when it is resized to a larger size, but I can't seem to do it. I also saw people doing this:
```
if event.type == pygame.VIDEORESIZE:
    window = pygame.display.set_mode((event.w, event.h), pygame.RESIZABLE)
```
but this just doesn't work for me. It breaks after I resize the window even a little.

Can someone help me find a solution to this? I'm sorry, if this is actually the correct behaviour and I'm just dumb. But if this is the expected behaviour, IMO resizing is broken.

**Comment by robertpfeiffer** [comment]
Resizing interacts weirdly with many window managers and can generate multiple events. If you change the window on the first resize event, you might make the window just one pixel bigger. The workaround is to resize the window once the resizing events have stopped, or to only use the last resizing event in a frame.

https://github.com/robertpfeiffer/yarn-pygame/blob/4de45d131a8ca8868c93b2c6357dc1d21448ad6d/editor.py#L249
https://github.com/robertpfeiffer/yarn-pygame/blob/4de45d131a8ca8868c93b2c6357dc1d21448ad6d/editor.py#L428 

cf. issues #201 and #1492

**Comment by SimonMaracine** [comment]
Thank you, but this workaround still doesn't work that well. After a short while resizing the window, it strangely stops generating VIDEORESIZE events (under Linux).

**Comment by nthykier** [comment]
Re: VIDEORESIZE events stopping all of sudden:  I think you are suffering from a bug in SDL1.2 as noted in the similar bug #201.

Re: resizing and window not updating to reflect that.  Interestingly, in SDL2, the window is automatically resized with the event.  Therefore I suspect (without having tested it) that your code would work in pygame 2 + SDL 2 - that is once we have found and implemented a solution to #1639.
